Major UI Refresh

lalitpatel opened the pull request:
## Goal
To refresh the UI of the App.

## Approach
* Use the [bulma](https://github.com/jgthms/bulma) css framework

## Changes & Enhancements

* Hide replay controls after the draft replay has ended.
* Optimize, consolidate and clean up images.
* Fix missing key errors for ReactJS.
* Make alerts optionally closable.
* Fix routing, back navigation and scrolling to top.
* Use better icons for Navbar.
* Use `aria-label` for buttons with images.

### Todo:

- [x] Test on IE and Firefox.
- [x] QA
- [x] Update tests (?)

### Screenshots:
<details><summary>
Show screenshots
</summary>
<p>

![Home Page](https://user-images.githubusercontent.com/13775/84567950-dac27500-ad99-11ea-8c16-f19ace37bfa4.png)

![Presets](https://user-images.githubusercontent.com/13775/84567968-f3328f80-ad99-11ea-95b0-7156e8ab6573.png)

![Watch](https://user-images.githubusercontent.com/13775/84567970-f75ead00-ad99-11ea-8f35-405e17b8424c.png)

![Draft](https://user-images.githubusercontent.com/13775/84568016-3ee53900-ad9a-11ea-8ec5-ed15d9fee35c.png)
</p>
</details>

HSZemi commented on src/components/menu/Index.tsx:
```diff
@@ -18,43 +18,45 @@ class Index extends React.Component<WithTranslation, IState> {
+                <div className="content box" id="instructions">
+                    <h3><Trans>instructions.title</Trans></h3>
+                    <p>
+                        <Trans i18nKey='instructions.1'>Captains mode is a turn-based civilization picker.
+                            Each
+                            captain can pick and ban civilizations in a predefined order. Bans
+                            prevent
+                            the opponent's captain from picking the civilizations.</Trans>
+                    </p>
+                    <p>
+                        <Trans i18nKey='instructions.2'>For each turn captains have <b>30 seconds</b>. In
+                            case
+                            of timeout a random civilization is picked for the captain or no civilization
+                            gets
+                            banned.</Trans>
+                    </p>
```
The newlines seem a bit choppy here

HSZemi commented on src/components/menu/ScrollToTop.js:
```diff
@@ -0,0 +1,16 @@
+import {useEffect} from "react";
```
Why is that a `js` file and not a `tsx` file?

HSZemi commented on src/components/menu/TopRightControls.tsx:
```diff
@@ -7,15 +7,33 @@ import IconStyleSelector from "../../containers/IconStyleSelector";
+            <nav className="navbar is-flex-touch" role="navigation" aria-label="main navigation">
+                <div className="navbar-brand">
+                    <h1 className="navbar-item">Captains Mode for Age of Empires II</h1>
+                    {/*<h2 className="subtitle">Captains Mode</h2>*/}
```
This looks like it can be removed

HSZemi commented on src/index.tsx:
```diff
@@ -4,16 +4,17 @@ import {Provider} from 'react-redux';
-import './style-material.css';
-import './style.css';
+import 'bulma/css/bulma.css'
+// import './pure-min.css';
+// import './style-material.css';
+// import './style.css';
```
Looks like those can be removed

HSZemi commented on src/index.tsx:
```diff
@@ -117,18 +119,20 @@ ReactDOM.render(
-                <Route path="/practice" component={Menu}/>
-                <Route component={NotFound404}/>
-            </Switch>
+            <ScrollToTop>
```
Is that one really necessary? It somehow feels nicer if the page does not jump back to the top when changing tabs…

lalitpatel replied:
I feel that is the user expected behaviour when you navigate to a different page. If we remove this code and navigate to spectate draft that is listed, for example, the user has scroll manually to the top. This is not happening in the current production version, but happens in the latest version. 

There is an alternative way to make the scroll animation smoother.

HSZemi replied:
Can we somehow get different behaviour for clicking on tabs and the other page transitions? :thinking:

lalitpatel replied:
It seems difficult at this point. I can either remove that functionality for now or we try to address selective scroll to top in subsequent updates.

HSZemi replied:
Let's keep the automatic scroll to top functionality out for now

HSZemi commented on src/models/Preset.ts:
```diff
@@ -7,7 +7,7 @@ class Preset {
     public static readonly EMPTY: Preset = new Preset('', [], []);
 
-    public static readonly NEW: Preset = new Preset('Change me', Civilisation.ALL, []);
+    public static readonly NEW: Preset = new Preset('', Civilisation.ALL, []);
```
I would like to have some text in there by default. It does not necessarily have to be "Change me".

lalitpatel replied:
I had added placeholder in the input box to take care of that. Is that sufficient?
![image](https://user-images.githubusercontent.com/13775/84584212-0dfb1780-ae1f-11ea-9ab7-62b57edad62c.png)

HSZemi replied:
No, because if you ignore the input box and click on the "create" button straight away, as it often happens, you get a draft without title that shows up in the latest drafts list like that as well.

HSZemi replied:
I guess it's either we 
- leave it like that, resulting sometimes in empty titles,
- make that a mandatory field,
- pre-fill it with a fixed value like "change me", "Noob Draft", …
- pre-fill it with a random value like "{adjective} Draft", where `adjective` comes from a list of "Amazing", "Adorable", "Marvellous", "Magnificient", …

lalitpatel replied:
Got it. I have reverted this and the current behaviour is:
> pre-fill it with a fixed value like "change me", "Noob Draft", …

We can address the validation here and also in "Enter code to Join Draft / Spectate" screens and "Chose name" modal in the subsequent PRs.

lalitpatel commented on src/components/PresetEditor/TurnExplanation.tsx:
```diff
@@ -2,56 +2,89 @@ import * as React from "react";
-                second turn).</p>
+            </li>
+            <li>A <span className="tag">NONEXCLUSIVE BAN</span> or <div className="tags has-addons"><span
+                className="tag has-text-weight-bold is-danger">N</span><span
+                className="tag has-text-weight-bold is-danger is-light">BAN</span></div>&nbsp;
```
Will address that in a separate PR.

lalitpatel commented:
I have updated the UI as per the suggestions in the PR:

- Decreased spacing in the top
- Make Draft screen width fluid 
- Centered the turn tags
- Centered the button in the draft message
- Reverted to classic icons for units / emblems

HSZemi commented:
I have another icon question: Why use different "random" icons for the random choice and for the random pick/ban/snipe indicator?
![grafik](https://user-images.githubusercontent.com/3027817/84597569-1d04b880-ae65-11ea-8136-01cb44da6887.png)

lalitpatel commented:
> I have another icon question: Why use different "random" icons for the random choice and for the random pick/ban/snipe indicator?

I have reverted the random icon to the original one because the one one did not match the over all theme of the civ chooser. I will scout for a better icon to replace it. For now, we can continue with the original icon.

HSZemi commented:
What do you dislike about the old random icon?

lalitpatel commented:
> What do you dislike about the old random icon?

I think the dice with `?` is a good concept but I felt that the actual image didnot go well with the theme. Also it looked a bit odd when it was overlayed over the cross. I was not able to find a similar image that would match the theme so settled for a regular dice.

HSZemi commented:
Looks like we cannot use that icon without paying though:
https://www.iconfinder.com/icons/2617623/chance_dice_draw_gambling_play_random_risk_icon

Maybe we can find something suitable on https://openclipart.org/search/?query=dice

lalitpatel commented:
Made the requested changes:
- removed scroll to top behaviour
- using a new dice image.

HSZemi commented:
I am confused: Even without the `ScrollToTop`, the page still jumps to the top when clicking on links :thinking: 
The `ScrollToTop` still is in the code btw. but it is unused

HSZemi commented:
@lalitpatel could you remove `src/components/menu/ScrollToTop.tsx`? I would then like to merge this :tada:

lalitpatel commented:
> I am confused: Even without the `ScrollToTop`, the page still jumps to the top when clicking on links 🤔
> The `ScrollToTop` still is in the code btw. but it is unused

Yes. Even I am confused now. 😕 
I have removed `ScrollToTop.tsx`
